sdk/storage/azure-storage-file-share/azure/storage/fileshare/_generated/_azure_file_storage.py

```diff
@@ -67,7 +68,24 @@ def __init__(  # pylint: disable=missing-client-constructor-parameter-credential
             allow_source_trailing_dot=allow_source_trailing_dot,
             **kwargs
         )
-        self._client: PipelineClient = PipelineClient(base_url=base_url, config=self._config, **kwargs)
+        _policies = kwargs.pop("policies", None)
```

Comment:
Can you verify we are passing our own `policies` in here (via breakpoint on client construction or code inspection). I assume we are but if not, this would likely be a breaking change for us to just get this default set of policies.

Reply:
![image](https://github.com/Azure/azure-sdk-for-python/assets/101599632/7f23d564-4015-4904-8336-09e8dd5d3a8a)
It would actually appear that we do _not_ pass in our own policy here 😮 

In our client code, constructing `AzureFileStorage` generally looks like this: https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/storage/azure-storage-file-share/azure/storage/fileshare/_share_service_client.py#L130-L133 (consistent across the package, none explicitly pass a `policies` argument)

Curiously enough though, we do pass `self._pipeline` but the constructor for AzureFileStorage seems to not even use that.. The plot thickens with ->

Our policies do get defined in base_client [here](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/storage/azure-storage-file-share/azure/storage/fileshare/_shared/base_client.py#L244), gets used for pipeline construction, then the pipeline variable is assigned [self._pipeline](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/storage/azure-storage-file-share/azure/storage/fileshare/_shared/base_client.py#L110)

SO it would seem that we don't override the policies here, but we do have policies defined, use them for a pipeline object construction, then pass this object to AzureFileStorage construction which seemingly does nothing with it¿
😕 😕 😕